### DISCUSSION TITLE
Spreading pageParams on the generator side

### DISCUSSION THREAD

**Discussion by vitorcamachoo** [orval-labs/orval#1598] [Ideas]
I've been checking the documentation to understand how pagination queries work through the generated process.
Sadly I found that for most of the founded use cases, the pageParam is being used as a Primitive and not as an object.

By using it as an object, we could have a generic approach and leave the responsibility to what getPreviousPageParam and getNextPageParam return, for example:
```tsx
initialPageParam: {
  ending_before: undefined as string | undefined,
},
getNextPageParam: (lastPage) => ({ ending_before: lastPage.events.at(-1)?.date }),

```

And on the generation part:
```tsx 
(example)
/gen/index.ts
 const queryFn: QueryFunction<Awaited<ReturnType<typeof getApiProjectsProjectIdEvents>>> = ({
    signal,
    pageParams = {}
  }) => getApiProjectsProjectIdEvents(projectId, {...params, ...pageParams}, requestOptions, signal);
```

**Comment by vitorcamachoo** [comment]
This request comes after an attempt of using only the `<scope>QueryOptions` generated methods as I can then choose what kind of query I want to use on the app side, meaning, useQuery, useSuspenseQuery, useSuspenseInfiniteQuery...

**Reply by melloware** [reply]
PR is welcome.
